README.md

```diff
@@ -1,17 +1,154 @@
+
+## Step 1: Set up the Monkey Policy
+
+### Latency
+
+```
+// context free setup
+var chaosPolicy = MonkeyPolicy.InjectLatency(
+	latency: TimeSpan.FromMinutes(1), // or randomized
+	injectionRate: 0.5, // inject randomly 50% of the time
+	enabled: () => isEnabled() // switch expt on/off by config
+	);
+	
+// using context to set up the policy
+var context = new Context();
+context["Enabled"] = isEnabled();
+context["InjectionRate"] = someInjectionRateProvider();
+context["ShouldInjectLatency"] = shouldInjectLatency();
+			
+Func<Context, bool> enabled = (ctx) =>
+{
+	// more validations here based on the context (ctx)
+	return ((bool)ctx["Enabled"]);
+};
+
+Func<Context, TimeSpan> latencyProvider = (ctx) =>
+{
+	// more validations here based on the context (ctx)
+	if ((bool)ctx["ShouldInjectLatency"])
+	{
+		return someLatencyProvider();
+	}
+
+	return TimeSpan.FromMilliseconds(0);
+};
+
+Func<Context, double> injectionRate = (ctx) =>
+{
+	// more validations here based on the context (ctx)
+	if (ctx["InjectionRate"] != null)
+	{
+		return (double)ctx["InjectionRate"];
+	}
+
+	return 0;
+};
+chaosPolicy = MonkeyPolicy.InjectLatency(latencyProvider, injectionRate, enabled);
+```
+
+### Fault
+```
+// context free setup
+var fault = new SocketException("Monkey fault exception");
+var faultPolicy = MonkeyPolicy.InjectFault<SocketException>(
+	fault, 
+	injectionRate: 0.5, // inject randomly 50% of the time
+	enabled: () => isEnabled() // switch expt on/off by config
+	);
+
+// using context to set up the policy
+var context = new Context();
+var failureMessage = "Failure Message";
+context["ShouldFail"] = true;
+context["Message"] = failureMessage;
+context["InjectionRate"] = 0.5;
+
+Func<Context, Exception> fault = (ctx) =>
+
+	// more validations here based on the context (ctx)
+	if (ctx["Message"] != null)
+	{
+		return new InvalidOperationException(ctx["Message"].ToString());
+	}
+
+	return new Exception();
+};
+
+Func<Context, double> injectionRate = (ctx) =>
+{
+	// more validations here based on the context (ctx)
+	if (ctx["InjectionRate"] != null)
+	{
+		return (double)ctx["InjectionRate"];
+	}
+
+	return 0;
+};
+
+Func<Context, bool> enabled = (ctx) =>
+{
+	// more validations here based on the context (ctx)
+	return ((bool)ctx["ShouldFail"]);
+};
+
+faultPolicy = MonkeyPolicy.InjectFault(fault, injectionRate, enabled);
+```
+
+### Behavior
+```
+
+// todo: add examples here
+
+```
+
+## Step 2: Execute the Monkey Policy
+
+```
+// executes the chaos policy directly
+chaosPolicy.Execute(() => someMethod());
+
+// executes the chaos policy using Context
+chaosPolicy.Execute((ctx) => someMethod(), context);
+
+// wrap the chaos policy using a PolicyWrap
+var policyWrap = Policy
+  .Wrap(fallbackPolicy, timeoutPolicy, chaosLatencyPolicy);
+policyWrap.Execute(() => someMethod())
+```
```

Comment:
It would be possible to add a paragraph here like this;

>It is usual to place the Simmy policy innermost in a PolicyWrap.  By placing the chaos policies innermost, they subvert the usual outbound call at the last minute, substituting their fault or adding extra latency.  The existing Polly policies - further out in the PolicyWrap - still apply, so you can test how the Polly resilience you have configured handles the chaos/faults injected by Simmy.

And maybe a note like this:

 >The above examples demonstrate how to execute through a Simmy policy directly, and how to include a Simmy policy in an individual PolicyWrap.  If your policies are configured by .NET Core DI at StartUp, for example via HttpClientFactory, there are also patterns which can configure Simmy into your app as a whole, at StartUp.  For examples, see the [Simmy Sample App](https://github.com/Polly-Contrib/Polly.Contrib.SimmyDemo_WebApi).